src/main/java/com/ibm/crypto/plus/provider/XDHPrivateKeyImpl.java

```diff
@@ -141,12 +141,12 @@ public XDHPrivateKeyImpl(OpenJCEPlusProvider provider, AlgorithmParameterSpec pa
             this.key = scalar.get();
         try {
             if (this.key == null) {
-                int pub_size = CurveUtil.getPublicCurveSize(curve);
+                int pub_size = CurveUtil.getCurveSize(curve);
                 this.xecKey = XECKey.generateKeyPair(provider.getOCKContext(), this.curve.ordinal(), pub_size);
             } else {
                 this.algid = CurveUtil.getAlgId(this.params.getName());
                 byte[] der = buildOCKPrivateKeyBytes();
-                int priv_size = CurveUtil.getPrivateCurveSize(curve);
+                int priv_size = CurveUtil.getDEREncodingSize(curve);
```

Comment:
Should we change the name of the variable to `encodingSize`?

Reply:
Yes.